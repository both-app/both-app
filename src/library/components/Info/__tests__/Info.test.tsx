import React from 'react'
import { render } from 'react-native-testing-library'
import * as Haptics from 'expo-haptics'

import { Info } from '../'

describe('Info', () => {
  afterEach(() => {
    jest.resetAllMocks()
  })

  test('should display the component with the texts and the right color', () => {
    const { toJSON } = render(
      <Info primary="title" secondary="subtitle" color="white" />
    )

    expect(toJSON()).toMatchSnapshot()
  })

  test('should have an haptic feedback when the props hide change', () => {
    const notificationAsyncMock = jest.spyOn(Haptics, 'notificationAsync')

    const props = {
      primary: 'title',
      secondary: 'subtitle',
      color: 'white',
      withHapticFeedback: true,
    }

    // @ts-ignore
    const { update } = render(<Info {...props} hide={true} />)
    // @ts-ignore
    update(<Info {...props} hide={false} />)

    expect(notificationAsyncMock).toHaveBeenCalled()
  })
})
