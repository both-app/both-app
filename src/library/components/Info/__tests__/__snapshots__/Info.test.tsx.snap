// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info should display the component with the texts and the right color 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 14,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
    testID="infoPrimaryText"
  >
    title
  </Text>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 14,
        "textAlign": "center",
      }
    }
    testID="infoSecondaryText"
  >
    subtitle
  </Text>
</View>
`;
